plugins/destination_testing.go

```diff
@@ -3,243 +3,241 @@ package plugins
-		panic(err)
-	}
-	boolColumn := &schema.Bool{
-		Bool:   true,
-		Status: schema.Present,
-	}
-	intColumn := &schema.Int8{
-		Int:    1,
-		Status: schema.Present,
-	}
-	floatColumn := &schema.Float8{
-		Float:  1.1,
-		Status: schema.Present,
-	}
-	uuidColumn := &schema.UUID{}
-	if err := uuidColumn.Set("00000000-0000-0000-0000-000000000001"); err != nil {
-		panic(err)
-	}
-	textColumn := &schema.Text{
-		Str:    "test",
-		Status: schema.Present,
-	}
-	byteaColumn := &schema.Bytea{
-		Bytes:  []byte{1, 2, 3},
-		Status: schema.Present,
-	}
-	textArrayColumn := &schema.TextArray{}
-	if err := textArrayColumn.Set([]string{"test1", "test2"}); err != nil {
-		panic(err)
-	}
-	intArrayColumn := &schema.Int8Array{}
-	if err := intArrayColumn.Set([]int8{1, 2}); err != nil {
-		panic(err)
-	}
-	timestampColumn := &schema.Timestamptz{
-		Time:   time.Now(),
-		Status: schema.Present,
-	}
-	jsonColumn := &schema.JSON{
-		Bytes:  []byte(`{"test": "test"}`),
-		Status: schema.Present,
-	}
-	uuidArrayColumn := &schema.UUIDArray{}
-	if err := uuidArrayColumn.Set([]string{"00000000-0000-0000-0000-000000000001", "00000000-0000-0000-0000-000000000002"}); err != nil {
-		panic(err)
-	}
-	inetColumn := &schema.Inet{}
-	if err := inetColumn.Set("192.0.2.1/24"); err != nil {
-		panic(err)
-	}
-	inetArrayColumn := &schema.InetArray{}
-	if err := inetArrayColumn.Set([]string{"192.0.2.1/24", "192.0.2.1/24"}); err != nil {
-		panic(err)
-	}
-	cidrColumn := &schema.CIDR{}
-	if err := cidrColumn.Set("192.0.2.1"); err != nil {
-		panic(err)
-	}
-	cidrArrayColumn := &schema.CIDRArray{}
-	if err := cidrArrayColumn.Set([]string{"192.0.2.1", "192.0.2.1"}); err != nil {
-		panic(err)
-	}
-	macaddrColumn := &schema.Macaddr{}
-	if err := macaddrColumn.Set("aa:bb:cc:dd:ee:ff"); err != nil {
-		panic(err)
-	}
-	macaddrArrayColumn := &schema.MacaddrArray{}
-	if err := macaddrArrayColumn.Set([]string{"aa:bb:cc:dd:ee:ff", "11:22:33:44:55:66"}); err != nil {
-		panic(err)
-	}
-
-	data := schema.CQTypes{
-		cqID,
-		cqParentID,
-		boolColumn,
-		intColumn,
-		floatColumn,
-		uuidColumn,
-		textColumn,
-		byteaColumn,
-		textArrayColumn,
-		intArrayColumn,
-		timestampColumn,
-		jsonColumn,
-		uuidArrayColumn,
-		inetColumn,
-		inetArrayColumn,
-		cidrColumn,
-		cidrArrayColumn,
-		macaddrColumn,
-		macaddrArrayColumn,
-	}
-
-	return data
+func getTestLogger(t *testing.T) zerolog.Logger {
+	t.Helper()
+	zerolog.TimeFieldFormat = zerolog.TimeFormatUnixMs
+	return zerolog.New(zerolog.NewTestWriter(t)).Output(
+		zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.StampMicro},
+	).Level(zerolog.DebugLevel).With().Timestamp().Logger()
 }
 
-func DestinationPluginTestHelper(ctx context.Context, p *DestinationPlugin, logger zerolog.Logger, spec specs.Destination) error {
+func (s *DestinationPluginTestSuite) destinationPluginTestWriteOverwrite(ctx context.Context, p *DestinationPlugin, logger zerolog.Logger, spec specs.Destination) error {
```

Comment:
Doesn't this test only `OverWriteDeleteStale`?

If I remove the code that handles `skipTestOverWriteDeleteStale` (both sections of it) the test fails.
I'm not sure who this tests only overwrite mode.

May I suggest splitting this suite to 2 distinct suites, 1 for overwrite mode and another for overwrite with delete stale?

Reply:
I added a struct with options instead of `with`